Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+					float openingAngle = LERP(0, m_DoorMoveTime, m_ClosedAngle, m_OpenAngle, m_DoorMoveTimer.GetElapsedSimTimeMS());
+
+					m_Door->SetJointPos(m_Pos + openingOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (openingAngle * GetFlipFactor()));
+
+					// Do the deep check to clear away any terrain debris when the door is moving but only after a little while so it doesn't take a chunk out of the ground before being snapped into place
+					if (m_DoorMoveTimer.IsPastSimMS(50)) { m_Door->DeepCheck(true); }
+				}
+				break;
+			case CLOSED:
+				m_Door->SetJointPos(m_Pos + m_ClosedOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+				m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_ClosedAngle * GetFlipFactor()));
+				break;
+			case CLOSING:
+				if (!m_DoorMoveSound.IsBeingPlayed()) { m_DoorMoveSound.Play(m_Pos); }
+
+				if (m_DoorMoveTimer.IsPastSimMS(m_DoorMoveTime)) {
+					m_ResetDefaultTimer.Reset();
+					
+					m_Door->SetJointPos(m_Pos + m_ClosedOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (m_ClosedAngle * GetFlipFactor()));
+
+					// Draw the material representation of the door to the terrain in the closed state
+					if (m_DrawMaterialLayerWhenClosed) { DrawDoorMaterial(); }
+
+					m_DoorMoveSound.Stop();
+					m_DoorMoveEndSound.Play(m_Pos);
+
+					m_DoorState = CLOSED;
+				} else {
+					// Find interpolated positions and angles for closing
+					Vector closingOffset(LERP(0, m_DoorMoveTime, m_OpenOffset.m_X, m_ClosedOffset.m_X, m_DoorMoveTimer.GetElapsedSimTimeMS()), LERP(0, m_DoorMoveTime, m_OpenOffset.m_Y, m_ClosedOffset.m_Y, m_DoorMoveTimer.GetElapsedSimTimeMS()));
+					// TODO: Make this work across rotation 0
+					float closingAngle = LERP(0, m_DoorMoveTime, m_OpenAngle, m_ClosedAngle, m_DoorMoveTimer.GetElapsedSimTimeMS());
+
+					m_Door->SetJointPos(m_Pos + closingOffset.GetXFlipped(m_HFlipped) * m_Rotation);
+					m_Door->SetRotAngle(m_Rotation.GetRadAngle() + (closingAngle * GetFlipFactor()));
+
+					// Do the deep check to clear away any terrain debris when the door is moving but only after a little while so it doesn't take a chunk out of the ground before being snapped into place
+					if (m_DoorMoveTimer.IsPastSimMS(50)) { m_Door->DeepCheck(true); }
+				}
+				break;
+			case STOPPED:
+				break;
+			default:
+				RTEAbort("Undefined door state value passed in. See DoorState enumeration for defined values.");
+				break;
+		}
+		m_Door->Update();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::Draw(BITMAP *targetBitmap, const Vector &targetPos, DrawMode mode, bool onlyPhysical) const {
+		// Override color drawing with flash, if requested.
+		DrawMode realMode = (mode == g_DrawColor && m_FlashWhiteMS) ? g_DrawWhite : mode;
+
+		if (m_Door && m_Door->IsAttached()) {
+			if (!m_Door->IsDrawnAfterParent()) {
```

Comment:
This could be trimmed down into two ternaries, though it might lower readability:
```
m_Door->IsDrawnAfterParent() ? m_Door->Draw(targetBitmap, targetPos, realMode, onlyPhysical) : Actor::Draw(targetBitmap, targetPos, mode, onlyPhysical);
m_Door->IsDrawnAfterParent() ? Actor::Draw(targetBitmap, targetPos, mode, onlyPhysical) : m_Door->Draw(targetBitmap, targetPos, realMode, onlyPhysical);
```

Reply:
also it would add an unnecessary evaluation of m_Door->IsDrawnAfterParent().

Reply:
Yeah, but that's basically the simplest possible evaluation - is X true. Probably takes a picosecond or something (I honestly have no idea how long, but it's very, very quick and shouldn't be worried about).
The readability question is the bigger problem, as the maxim (not to be confused with maximdude) goes, "Write for people, not the machine".

Reply:
Yeah i'm just gonna keep it as is.